strawberryfields/backends/states.py

```diff
@@ -689,6 +744,124 @@ def quad_expectation(self, mode, phi=0, **kwargs):
 
         return mean, var
 
+    def poly_quad_expectation(self, A, d=None, k=0, phi=0, **kwargs):
+        if A is None:
+            A = np.zeros([2*self._modes, 2*self._modes])
+
+        if A.shape != (2*self._modes, 2*self._modes):
+            raise ValueError("Matrix of quadratic coefficients A must be of size 2Nx2N.")
+
+        if np.any(A.T != A):
+            raise ValueError("Matrix of quadratic coefficients A must be symmetric.")
+
+        if d is None:
+            linear_coeff = np.zeros([2*self._modes])
+        else:
+            linear_coeff = d.copy()
+            linear_coeff[self._modes:] = -d[self._modes:]
```

Comment:
Couldn't understand what this line is for

Reply:
This actually really confuses me - for some reason, the einsum Fock calculation doesn't provide the same answers as the Guassian backend/theoretical result _unless_ I first modify `d` and `A` such that

1. The linear coefficients on the momentum operators are negated: `d' = (d_1,...,d_n,-d_{n+1},...,-d_{2n})`, and

2. The block-off diagonal elements of `A` (i.e. so the coefficients of the `x_np_m` terms) are also negated.

The only reason I know that this works is because I had the _exact_ same issue in OpenBoson, and this is the trick that allowed the Fock backends to provide the correct result.

Reply:
Perhaps @nquesada knows why there is a discrepency between backends? Is this something that the SF frontend compensates for?

Reply:
@nquesada didn't know either